src/components/TodoItem.tsx

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import { useState } from 'react';
+import { Todo } from '../types/Todo';
+import cn from 'classnames';
+import { USER_ID } from '../api/todos';
+
+const classNameLoader = 'modal-background has-background-white-ter';
+
+type Props = {
+  todo: Todo;
+  selectedTodo?: Todo;
+  setSelectedTodo: React.Dispatch<React.SetStateAction<Todo | undefined>>;
+  deleteChosenTodo: (currentTodoToDelete: Todo | null) => void;
+  focusedTodoRef: React.RefObject<HTMLInputElement>;
+  shouldDeleteCompleted: boolean;
+  updateChosenTodo: (todoSetToUpdate: Todo | null) => void;
+  shouldToggleAllCompleted: boolean;
+  currentTodos: Todo[];
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  selectedTodo,
+  setSelectedTodo,
+  deleteChosenTodo,
+  focusedTodoRef,
+  shouldDeleteCompleted,
+  updateChosenTodo,
+  shouldToggleAllCompleted,
+  currentTodos,
+}) => {
+  const [inputValue, setInputValue] = useState(todo.title);
+  const [loader, setLoader] = useState(false);
+  const currentToggle = currentTodos.some(toDo => !toDo.completed);
+
+  const handleDelete = async (todoToDel: Todo) => {
+    setLoader(true);
+
+    await new Promise(resolve => setTimeout(resolve, 100));
+
+    await deleteChosenTodo(todoToDel);
+
+    setLoader(false);
+  };
+
+  const handleUpdate = async (todoToUp: Todo) => {
+    setLoader(true);
+
+    await new Promise(resolve => setTimeout(resolve, 100));
+
+    await updateChosenTodo(todoToUp);
+
+    setLoader(false);
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(e.target.value);
+  };
+
+  const handleBlurOrSubmit = () => {
+    if (inputValue !== todo.title) {
+      if (inputValue === '') {
+        handleDelete(todo);
+      } else {
+        handleUpdate({
+          id: todo.id,
+          title: inputValue.trim(),
+          userId: USER_ID,
+          completed: todo.completed,
+        });
+      }
+    }
+
+    if (inputValue === todo.title) {
+      setSelectedTodo(undefined);
+    }
```

Comment:
```suggestion
    if (inputValue === todo.title) {
      setSelectedTodo(null);
      
      return;
    }
    
    if (inputValue) {
    // other code
```
You can add early return to avoid nested `if` statements.
Also, it's better to use `null` instead of `undefined` in this case